### PR TITLE
Enabling to create storage class using storage class with user input like provisoner and binding mode

### PR DESCRIPTION
**Pull request by k-shiva-sai**

### CODE REVIEW

File: src/oadp_resources/storage_class.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+
+from ocp_resources.resource import Resource
```
Comment (amastbau) [inline src/oadp_resources/storage_class.py:3]: Can you please create a storage folder and place it there?

**Review by amastbau — CHANGES_REQUESTED** [review]

File: src/oadp_resources/storage_class.py (hunk 1 above)
Comment (k-shiva-sai) [reply src/oadp_resources/storage_class.py:3]: OKAY Amos

File: src/oadp_resources/storage_class/storage_class.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                provisioner=self.provisioner,
+                volume_binding_mode=self.volume_binding_mode,
+            )
+
+    def add_provisioner(self,provisioner=None, volume_binding_mode=None):
+        if not self.res:
+            super().to_dict()
+        self.res["provisioner"]=provisioner
+        if volume_binding_mode:
+            self.res["volumeBindingMode"]=volume_binding_mode
```
Comment (amastbau) [inline src/oadp_resources/storage_class/storage_class.py:47]: Is this really needed if we set it with the constructor?
also, the name is confusing,  
it is setting a provider, not adding one.
wdyt?

**Review by amastbau — CHANGES_REQUESTED** [review]

File: src/oadp_resources/storage_class/storage_class.py (hunk 2 above)
Comment (amastbau) [reply src/oadp_resources/storage_class/storage_class.py:47]: maybe change the function name to set (it is setting the provider, not adding ine, correct?)

Comment (k-shiva-sai) [reply src/oadp_resources/storage_class/storage_class.py:47]: haa correct Amos I will update the commit in a moment

Comment (k-shiva-sai) [reply src/oadp_resources/storage_class/storage_class.py]: Modified Amos please give it a check

File: src/oadp_resources/storage_class/storage_class.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+        if not self.yaml_file and self.provisioner:
+            self.add_provisioner(
+                provisioner=self.provisioner,
+                volume_binding_mode=self.volume_binding_mode,
+            )
+
+    def set_provisioner(self,provisioner=None, volume_binding_mode=None):
+        if not self.res:
+            super().to_dict()
```
Comment (amastbau) [inline src/oadp_resources/storage_class/storage_class.py:44]: Please note,
that if using a yaml file to init the object
to_dict() returns this yaml as is

**Review by amastbau — APPROVED** [review]
Please merge only after testing.

**Pull request merged**